core/src/main/java/edu/wpi/grip/core/operations/composite/PublishVideoOperation.java

```diff
@@ -46,110 +59,55 @@
-        logger.info("Starting camera server");
-
-        try (Socket socket = serverSocket.accept()) {
-          logger.info("Got connection from " + socket.getInetAddress());
-          connected = true;
-
-          DataOutputStream socketOutputStream = new DataOutputStream(socket.getOutputStream());
-          DataInputStream socketInputStream = new DataInputStream(socket.getInputStream());
-
-          byte[] buffer = new byte[128 * 1024];
-          int bufferSize;
-
-          final int fps = socketInputStream.readInt();
-          final int compression = socketInputStream.readInt();
-          final int size = socketInputStream.readInt();
-
-          if (compression != -1) {
-            logger.warning("Dashboard video should be in HW mode");
-          }
-
-          final long frameDuration = 1000000000L / fps;
-          long startTime = System.nanoTime();
-
-          while (!socket.isClosed() && !Thread.currentThread().isInterrupted()) {
-            // Wait for the main thread to put a new image. This happens whenever perform() is
-            // called with
-            // a new input.
-            synchronized (imageLock) {
-              imageLock.wait();
-
-              // Copy the image data into a pre-allocated buffer, growing it if necessary
-              bufferSize = imagePointer.limit();
-              if (bufferSize > buffer.length) {
-                buffer = new byte[imagePointer.limit()];
-              }
-              imagePointer.get(buffer, 0, bufferSize);
-            }
-
-            // The FRC dashboard image protocol consists of a magic number, the size of the image
-            // data,
-            // and the image data itself.
-            socketOutputStream.write(MAGIC_NUMBER);
-            socketOutputStream.writeInt(bufferSize);
-            socketOutputStream.write(buffer, 0, bufferSize);
-
-            // Limit the FPS to whatever the dashboard requested
-            int remainingTime = (int) (frameDuration - (System.nanoTime() - startTime));
-            if (remainingTime > 0) {
-              Thread.sleep(remainingTime / 1000000, remainingTime % 1000000);
-            }
-
-            startTime = System.nanoTime();
-          }
-        }
-      } catch (IOException e) {
-        logger.log(Level.WARNING, e.getMessage(), e);
-      } catch (InterruptedException e) {
-        Thread.currentThread().interrupt(); // This is really unnecessary since the thread is
-        // about to exit
-        logger.info("Shutting down camera server");
-        return;
-      } finally {
-        connected = false;
-      }
-    }
-  };
+  private final MjpegServer server;
+  private final CvSource serverSource;
+
+  // Write to the /CameraPublisher table so the MJPEG streams are discoverable by other
+  // applications connected to the same NetworkTable server (eg Shuffleboard)
+  private static final ITable cameraPublisherTable = NetworkTable.getTable("/CameraPublisher");
```

Comment:
Can you not load this statically? Instead do it at instantiation time.

Reply:
What would that gain?

Reply:
If it fails then GRIP doesn't crash immediately.

Reply:
So instead of crashing to desktop at initialization when creating the operations list, it'll crash to desktop when adding a new PublishVideo operation. I guess that's better because at least users will see it happens when adding the operation